src/loadr.h

```diff
@@ -0,0 +1,278 @@
+macro(CharacterMode) \
+macro(GEaddDevice2) \
+macro(GEaddDevice2f) \
+macro(GEcopyDisplayList) \
+macro(GEcreateDevDesc) \
+macro(GEcreateSnapshot) \
+macro(GEgetDevice) \
+macro(GEkillDevice) \
+macro(GEplayDisplayList) \
+macro(GEplaySnapshot) \
+macro(get_R_HOME) \
+macro(getDLLVersion) \
+macro(getRUser) \
+macro(in_memsize) \
+macro(INTEGER) \
+macro(LOGICAL) \
+macro(PRCODE) \
+macro(PRVALUE) \
+macro(R_BaseEnv) \
+macro(R_CHAR) \
+macro(R_CheckDeviceAvailable) \
+macro(R_CleanUp) \
+macro(R_common_command_line) \
+macro(R_curErrorBuf) \
+macro(R_DefParams) \
+macro(R_EmptyEnv) \
+macro(R_FunTab) \
+macro(R_GE_getVersion) \
+macro(R_getEmbeddingDllInfo) \
+macro(R_GlobalContext) \
+macro(R_GlobalEnv) \
+macro(R_interrupts_pending) \
+macro(R_interrupts_suspended) \
+macro(R_IsNA) \
+macro(R_lsInternal3) \
+macro(R_NaInt) \
+macro(R_NamesSymbol) \
+macro(R_NaString) \
+macro(R_new_custom_connection) \
+macro(R_NilValue) \
+macro(R_ParseVector) \
+macro(R_PreserveObject) \
+macro(R_ProcessEvents) \
+macro(R_registerRoutines) \
+macro(R_ReleaseObject) \
+macro(R_RestoreGlobalEnvFromFile) \
+macro(R_SaveGlobalEnvToFile) \
+macro(R_set_command_line_arguments) \
+macro(R_SetParams) \
+macro(R_setStartTime) \
+macro(R_Srcref) \
+macro(R_ToplevelExec) \
+macro(R_UnboundValue) \
+macro(R_WaitEvent) \
+macro(RAW) \
+macro(RDEBUG) \
+macro(readconsolecfg) \
+macro(REAL) \
+macro(Rf_allocList) \
+macro(Rf_allocVector) \
+macro(Rf_allocVector3) \
+macro(Rf_asChar) \
+macro(Rf_asInteger) \
+macro(Rf_asLogical) \
+macro(Rf_asReal) \
+macro(Rf_classgets) \
+macro(Rf_curDevice) \
+macro(Rf_deparse1line) \
+macro(Rf_desc2GEDesc) \
+macro(Rf_duplicate) \
+macro(Rf_error) \
+macro(Rf_eval) \
+macro(Rf_findVar) \
+macro(Rf_getAttrib) \
+macro(Rf_install) \
+macro(Rf_installChar) \
+macro(Rf_isEnvironment) \
+macro(Rf_isNull) \
+macro(Rf_isString) \
+macro(Rf_length) \
+macro(Rf_mkChar) \
+macro(Rf_mkCharCE) \
+macro(Rf_mkString) \
+macro(Rf_ndevNumber) \
+macro(Rf_NewEnvironment) \
+macro(Rf_onintr) \
+macro(Rf_protect) \
+macro(Rf_ScalarInteger) \
+macro(Rf_ScalarLogical) \
+macro(Rf_ScalarReal) \
+macro(Rf_ScalarString) \
+macro(Rf_selectDevice) \
+macro(Rf_translateCharUTF8) \
+macro(Rf_unprotect) \
+macro(Rf_utf8towcs) \
+macro(Rf_wtransChar) \
+macro(run_Rmainloop) \
+macro(SET_RDEBUG) \
+macro(SET_STRING_ELT) \
+macro(SET_TYPEOF) \
+macro(SET_VECTOR_ELT) \
+macro(SETCAR) \
+macro(setup_Rmainloop) \
+macro(STRING_ELT) \
+macro(TYPEOF) \
+macro(VECTOR_ELT) \
+macro(vmaxget) \
+macro(vmaxset) \
+
+#ifdef _WIN32
+#define _RGRAPHAPPAPI_SET(macro) \
+macro(GA_initapp) 
+#endif
+
+namespace rhost {
+    namespace rapi {
+
+        // R.dll/R.so APIs
+        _RAPI_SET(_RAPI_DECL_END);
+
+#ifdef _WIN32
+        // Rgraphapp.dll/Rgraphapp.so APIs
+        _RGRAPHAPPAPI_SET(_RAPI_DECL_END);
+#endif
+        void load_r_apis(fs::path r_dll_dir);
+        void unload_r_apis();
+    }
+}
+
+#ifndef RHOST_NO_API_REDIRECT
+#define CAR rhost::rapi::_RAPI_PTR(CAR)
```

Comment:
Do these have to be macros? It feels like if you drop the `fp_` prefix in `_RAPI_PTR`, then we'll just have variables with the right names directly in the namespace. And because those variables will have function pointer types, they will be callable directly. You'll only need to do `using namespace rhost::rapi` in other files.